ui/src/app/workflows/components/submit-workflow-panel.tsx

```diff
@@ -17,119 +17,92 @@ interface Props {
 
-    private getSelectedTemplate(entrypoint: string): Template {
-        for (const t of this.state.templates) {
-            if (t.name === entrypoint) {
-                return t;
-            }
-        }
-        return null;
-    }
+    const getSelectedTemplate = (name: string): Template | null => {
+        return templates.find(t => t.name === name) || null;
+    };
 
-    private submit() {
-        this.setState({isSubmitting: true});
+    const submit = () => {
+        setIsSubmitting(true);
         services.workflows
-            .submit(this.props.kind, this.props.name, this.props.namespace, {
-                entryPoint: this.state.entrypoint === workflowEntrypoint ? null : this.state.entrypoint,
+            .submit(props.kind, props.name, props.namespace, {
+                entryPoint: entrypoint === workflowEntrypoint ? null : entrypoint,
                 parameters: [
-                    ...this.state.workflowParameters.filter(p => Utils.getValueFromParameter(p) !== undefined).map(p => p.name + '=' + Utils.getValueFromParameter(p)),
-                    ...this.state.parameters.filter(p => Utils.getValueFromParameter(p) !== undefined).map(p => p.name + '=' + Utils.getValueFromParameter(p))
+                    ...workflowParameters.filter(p => Utils.getValueFromParameter(p) !== undefined).map(p => p.name + '=' + Utils.getValueFromParameter(p)),
+                    ...parameters.filter(p => Utils.getValueFromParameter(p) !== undefined).map(p => p.name + '=' + Utils.getValueFromParameter(p))
                 ],
-                labels: this.state.labels.join(',')
+                labels: labels.join(',')
             })
             .then((submitted: Workflow) => (document.location.href = uiUrl(`workflows/${submitted.metadata.namespace}/${submitted.metadata.name}`)))
-            .catch(error => this.setState({error, isSubmitting: false}));
-    }
+            .catch(err => {
+                setError(err);
+                setIsSubmitting(false);
+            });
+    };
+
+    return (
+        <>
+            <h4>Submit Workflow</h4>
+            <h5>
+                {props.namespace}/{props.name}
+            </h5>
+            {error && <ErrorNotice error={error} />}
+            <div className='white-box'>
+                <div key='entrypoint' title='Entrypoint' style={{marginBottom: 25}}>
+                    <label>Entrypoint</label>
+                    <Select
+                        value={entrypoint}
+                        options={templates.map(t => ({
+                            value: t.name,
+                            title: t.name
+                        }))}
+                        onChange={selected => {
+                            const selectedTemp = getSelectedTemplate(selected.value);
+                            setEntrypoint(selected.value);
+                            setParameters((selectedTemp && selectedTemp.inputs.parameters) || []);
```

Comment:
```suggestion
                            setParameters(selectedTemp?.inputs?.parameters || []);
```

can simplify this with optional chaining syntax